sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -444,138 +441,98 @@
+     * {@code label} combination. If the {@code isReadOnly} is true or null, the setting will be set to read-only.
+     * If false, the setting will not be read-only.
      *
      * <p><strong>Code Samples</strong></p>
      *
      * <p>Set the setting to read-only with the key-label "prodDBConnection"-"westUS".</p>
      *
-     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string}
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string-boolean}
+     *
+     * <p>Clear read-only of the setting with the key-label "prodDBConnection"-"westUS".</p>
+     *
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string-boolean-clearReadOnly}
      *
      * @param key The key of configuration setting to set to be read-only.
-     * @param label The label of configuration setting to read-only, or optionally, null if a setting with label is
-     * desired.
+     * @param label The label of configuration setting to read-only. If {@code null} no label will be used.
+     * @param isReadOnly The boolean value to set the setting to read-only or not read-only.
+     * If it is true, set the setting to read-only. If false, the setting won't set to read-only.
      * @return The {@link ConfigurationSetting} that is read-only, or an empty Mono if a key collision occurs or the
      * key is an invalid value (which will also throw HttpResponseException described below).
      * @throws IllegalArgumentException If {@code key} is {@code null}.
      * @throws HttpResponseException If {@code key} is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<ConfigurationSetting> setReadOnly(String key, String label) {
+    public Mono<ConfigurationSetting> setReadOnly(String key, String label, boolean isReadOnly) {
         try {
             return withContext(context -> setReadOnly(
-                new ConfigurationSetting().setKey(key).setLabel(label), context))
+                new ConfigurationSetting().setKey(key).setLabel(label), isReadOnly, context))
                 .flatMap(response -> Mono.justOrEmpty(response.getValue()));
         } catch (RuntimeException ex) {
             return monoError(logger, ex);
         }
     }
 
     /**
-     * Set the {@link ConfigurationSetting} to read-only with a matching {@link ConfigurationSetting#getKey() key},
-     * and optional {@link ConfigurationSetting#getLabel() label} combination.
+     * Set the {@link ConfigurationSetting} to read-only or not read-only with a matching
+     * {@link ConfigurationSetting#getKey() key}, and optional {@link ConfigurationSetting#getLabel() label}
+     * combination. If the {@code isReadOnly} is true or null, the setting will be set to read-only.
+     * If false, the setting will not be read-only.
```

Comment:
Same as before, this is the leading paragraph and should simply state what the API is used for, setting the read-only flag on the configuration.